### ISSUE TITLE
Set up long-term ForTrilinos integration with Trilinos

### ISSUE THREAD

**Issue by sethrj** [trilinos/ForTrilinos#270]
- [x] Replace ForTrilinos TriBITS build system with modern CMake build and a
      `find_package(Trilinos)` dependency.
- [x] Update ForTrilinos continuous integration to use the tagged versions of
      SWIG-Fortran and Trilinos.
- [x] Determine ForTrilinos versioning scheme
- [x] Add Spack package and git release numbers for ForTrilinos.
- [x] Remove ForTrilinos from Trilinos "experimental external repository" list and
      Spack package (https://github.com/trilinos/Trilinos/pull/8039, https://github.com/spack/spack/pull/18456)
- [x] Coordinate with E4S for distribution. (https://github.com/xsdk-project/xsdk-policy-compatibility/pull/27, https://github.com/E4S-Project/e4s/pull/5)

## Versioning scheme

From a downstream app's perspective, the main issue will probably be "what works with my interface to the fortrilinos version" vs "what works in my other C++ code in this project for a given trilinos version". Since ForTrilinos is tightly coupled to both Trilinos and SWIG-Fortran as dependencies, major changes to either one will cause major changes to the ForTrilinos interface and the downstream app's code.

Conceivably there will be ForTrilinos-only codes whose only concern is the ForTrilinos API, so I'd argue that should be the primary factor in versioning.

So I propose using [SemVer 2.0](https://semver.org):
> Bug fixes not affecting the API increment the patch version, backwards compatible API additions/changes increment the minor version, and backwards incompatible API changes increment the major version.
With roughly the following practical translation with respect to SWIG and Trilinos:
- Major version numbers with Trilinos and minor versions of SWIG-Fortran (since it's still not officially upstreamed) *can* result in major version number changes for ForTrilinos
- New features in Trilinos, and new support by ForTrilinos, *can* result in minor version number changes
- Minor changes to the SWIG-Fortran implementation (which don't affect the interface in the `.F90` files) result in a patch version.
Basically, the versioning will be driven by what the Fortran-only users see in the committed version of the generated wrappers.

We will keep a very visible table synchronized with the ForTrilinos spack package that codifies "known working" version compatibility, e.g.:

ForTrilinos | Trilinos | SWIG-Fortran
----- | ---- | ----
0.5.0 | 12.17–12.18.0 | 15e6ed597386e203d0348d94c308e178b80ec5bb

**Comment by sethrj** [comment]
@aprokop I'd like to decide on a versioning scheme ASAP so I can get the spack package set up with a version number. A secondary question (if we go with an independent SemVer) is whether we backdate previous "releases" of ForTrilinos, even though they're unable to be built independently.

**Comment by aprokop** [comment]
Agreed to follow semantic versioning.
